Stratis.Bitcoin/Logging/Logs.cs

```diff
@@ -8,56 +8,149 @@ namespace Stratis.Bitcoin.Logging
+			LoggerFactory = factory;
+
+			// These match namespace; classes can also use CreateLogger<T>, which will inherit
+			Configuration = factory.CreateLogger("Stratis.Bitcoin.Configuration");
+			RPC = factory.CreateLogger("Stratis.Bitcoin.RPC");
+			FullNode = factory.CreateLogger("Stratis.Bitcoin.FullNode");
+			ConnectionManager = factory.CreateLogger("Stratis.Bitcoin.Connection");
+			Bench = factory.CreateLogger("Stratis.Bitcoin.FullNode.ConsensusStats");
+			Mempool = factory.CreateLogger("Stratis.Bitcoin.MemoryPool");
+			BlockStore = factory.CreateLogger("Stratis.Bitcoin.BlockStore");
+			EstimateFee = factory.CreateLogger("Stratis.Bitcoin.Fee");
 		}
 
-		public static void Configure(ILoggerFactory factory)
+		public static ILoggerFactory GetLoggerFactory(string[] args)
 		{
-			Configuration = factory.CreateLogger("Configuration");
-			RPC = factory.CreateLogger("RPC");
-			FullNode = factory.CreateLogger("FullNode");
-			ConnectionManager = factory.CreateLogger("ConnectionManager");
-			Bench = factory.CreateLogger("Bench");
-			Mempool = factory.CreateLogger("Mempool");
-			BlockStore = factory.CreateLogger("BlockStore");
-			EstimateFee = factory.CreateLogger("EstimateFee");
+			// TODO: preload enough args for -conf= or -datadir= to get debug args from there
+
+			//Configuration = factory.CreateLogger("Configuration");
+			//FullNode = factory.CreateLogger("FullNode");
+			//ConnectionManager = factory.CreateLogger("ConnectionManager");
+			//EstimateFee = factory.CreateLogger("EstimateFee");
+
+			var debugArgs = args.Where(a => a.StartsWith("-debug=")).Select(a => a.Substring("-debug=".Length).Replace("\"", "")).FirstOrDefault();
+			var keyToCategory = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+				{
+					//{ "addrman", "" },
+					//{ "alert", "" },
+					{ "bench", "Stratis.Bitcoin.FullNode.ConsensusStats" },
+					//{ "coindb", "" },
+					{ "db", "Stratis.Bitcoin.BlockStore" }, 
+					//{ "lock", "" }, 
+					//{ "rand", "" }, 
+					{ "rpc", "Stratis.Bitcoin.RPC" }, 
+					//{ "selectcoins", "" }, 
+					{ "mempool", "Stratis.Bitcoin.MemoryPool" }, 
+					//{ "mempoolrej", "" }, 
+					{ "net", "Stratis.Bitcoin.Connection" }, 
+					//{ "proxy", "" }, 
+					//{ "prune", "" }, 
+					//{ "http", "" }, 
+					//{ "libevent", "" }, 
+					//{ "tor", "" }, 
+					//{ "zmq", "" }, 
+					//{ "qt", "" },
+
+					// Short Names
+					{ "estimatefee", "Stratis.Bitcoin.Fee" },
+					{ "configuration", "Stratis.Bitcoin.Configuration" },
+					{ "fullnode", "Stratis.Bitcoin.FullNode" },
+				};
+			var filterSettings = new FilterLoggerSettings();
+			// Default level is Information
+			filterSettings.Add("Default", LogLevel.Information);
+			// TODO: Probably should have a way to configure these as well
+			filterSettings.Add("System", LogLevel.Warning);
+			filterSettings.Add("Microsoft", LogLevel.Warning);
+			//Disable aspnet core logs
+			filterSettings.Add("Microsoft.AspNetCore", LogLevel.Error);
```

Comment:
Yes. The setting is copied from Startup.cs, so is exactly replicating the previous behaviour (rather than a second logging factory in ASP.NET, I pass in a single factory).

The existing system was configured to log Error & above for AspNetCore, so I kept that the same. (I think Warning & above is probably better; the previous config only changed Microsoft.AspNetCore, whereas I also set the rest of System & Microsoft).